optimum/intel/openvino/modeling.py

```diff
@@ -0,0 +1,259 @@
+)
+from transformers.file_utils import add_start_docstrings, add_start_docstrings_to_model_forward
+from transformers.modeling_outputs import QuestionAnsweringModelOutput, SequenceClassifierOutput, TokenClassifierOutput
+
+from .modeling_base import OVBaseModel
+
+
+logger = logging.getLogger(__name__)
+
+
+_TOKENIZER_FOR_DOC = "AutoTokenizer"
+
+MODEL_START_DOCSTRING = r"""
+    This model inherits from [~`intel.openvino.modeling.OVBaseModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving)
+    Parameters:
+        config (`transformers.PretrainedConfig`): [PretrainedConfig](https://huggingface.co/docs/transformers/main_classes/configuration#transformers.PretrainedConfig) is the Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~intel.openvino.modeling.OVBaseModel.from_pretrained`] method to load the model weights.
+        model (`openvino.pyopenvino.Model`): is the main class used to run OpenVINO Runtime inference.
+"""
+
+INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (`torch.Tensor`):
+            Indices of input sequence tokens in the vocabulary.
+            Indices can be obtained using [`AutoTokenizer`](https://huggingface.co/docs/transformers/autoclass_tutorial#autotokenizer).
+            [What are input IDs?](https://huggingface.co/docs/transformers/glossary#input-ids)
+        attention_mask (`torch.Tensor`)`, *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+            [What are attention masks?](https://huggingface.co/docs/transformers/glossary#attention-mask)
+        token_type_ids (`torch.Tensor`, *optional*):
+            Segment token indices to indicate first and second portions of the inputs. Indices are selected in `[0, 1]`:
+            - 1 for tokens that are **sentence A**,
+            - 0 for tokens that are **sentence B**.
+            [What are token type IDs?](https://huggingface.co/docs/transformers/glossary#token-type-ids)
+"""
+
+
+class OVModel(OVBaseModel):
+
+    base_model_prefix = "ov_model"
+    auto_model_class = AutoModel
+
+    def __init__(self, model=None, config=None, **kwargs):
+        super().__init__(model, config, **kwargs)
+        # Avoid warnings when creating a transformers pipeline
+        AutoConfig.register(self.base_model_prefix, AutoConfig)
+        self.auto_model_class.register(AutoConfig, self.__class__)
+        self.device = torch.device("cpu") if self._device == "CPU" else torch.device("cuda")
+
+    def to(self, device):
+        self.device = device
+        self._device = "CPU" if device == torch.device("cpu") else "GPU"
```

Comment:
This should not be connected to "torch.device()". OpenVINO has its own set of supported devices which is independent of what is available in PyTorch.

Reply:
I suggest changing it to:
```suggestion
        self._device = device
```

Reply:
My bad, I was thinking you were talking about `OVBaseModel`. `OVBaseModel` has no dependency on torch but `OVModel` does (and the `forward` method of every `OVModelForXxxx` will have torch tensors as inputs and outputs), so we should keep it as it is in my opinion

Reply:
My opinion is that we should keep the inference device independent from the input/output torch.Tensor device type. @echarlaix, can you please explain how we can use Intel GPUs for inference which is one of the benefit of using OpenVINO?

Reply:
I changed the code in order for both devices to be independent